Add Jetpack module search capability

nickpagz opened the pull request:
### Changes Proposed in this Pull Request
Add a new command that allows a search for a specific module status and returns a list of sites with that status.
In other words, "Find all sites where the `wordads` module is currently `on`".

Note: Sites must have a healthy Jetpack connection for this to work.

### Testing Instructions

- From your `team51-cli` folder in a new terminal window, switch to this branch using `git checkout feature/jetpack-module-search`.
- Test the command on your favorite Team 51 managed site, eg. `team51 jetpack-sites-with adwords on`, or, `team51 jetpack-sites-with infinite-scroll off`.
- Be warned, this process takes about 8 to 10 minutes to complete.
- As an output you should see two lists - the first is the list of sites with the module status, and the second a list of sites that could not be queried to a Jetpack connection issue.

ecairol commented on src/commands/jetpack-module.php:
```diff
@@ -0,0 +1,118 @@
+		$sites_not_checked   = array();
+		foreach ( $site_list as $site ) {
+			$progress_bar->advance();
+			$module_list = $this->get_list_of_modules( $site[0] );
```
I wonder if the whole process would speed up if we make this call async. 

We do something like that for the `remove-user` command. However, I'm in the process of looking for a better async option because the one we use (`Amphp::parallelMap`) sometimes doesn't work on all computers.

nickpagz replied:
That would be helpful since it does take a very long time. It's also the same issue with #75 since it uses the same process. 
Does this then just run that particular function in multiple CPU threads? I'm curious as to how it handles the multiple calls, and could that create issues with API rate limits?

ecairol replied:
> Does this then just run that particular function in multiple CPU threads? 

I think so, but not 100% sure. Sounds like at some point they require to create temporary files on the fly, and that triggers an error on some computers.

This is their main repository: https://github.com/amphp

We use Parallel, but they also have a library dedicated to Async HTTP requests which I plan to give a try.

nickpagz replied:
Ok. I think I'll merge as it is for now, along with the command renaming, and we can tackle this later. 
Noting here, I did try several options to speed this up, similar to what you discovered [here](https://github.com/Automattic/team51-cli/pull/77#discussion_r875050753), however it doesn't look like I can filter down through the fields with the `jetpack-blogs` endpoint. 
I'm thinking the delay is happening because of the sequential API calls, one after the other for each site. I was hoping I would have been able to download all of the sites' data in one call, then do the search/comparison on the downloaded data - but as far as I can tell it's not possible without looping through each site ID.

ecairol replied:
> Ok. I think I'll merge as it is for now

Yes, I agree that's a good call.

> I'm thinking the delay is happening because of the sequential API calls, one after the other for each site.

I think the same. This week I'll be working on an implementation for async calls (different than `parallelMap`), I'll ping you when we have a function that you can re-use here @nickpagz

nickpagz commented on src/commands/jetpack-module.php:
```diff
@@ -0,0 +1,118 @@
+
+
+class Jetpack_Module_Search extends Command {
+	protected static $defaultName = 'module-search';
```
I'm scratching my head on this one... how did I even test that? 
I think I made the change at the last minute, maybe because it's too close to `jetpack-modules` in #78 
The other option is to rename both, so like `jetpack-module-list` and `jetpack-module-search` - longer, but more explicit. What do you think?

ecairol replied:
I think `jetpack-module-list` and `jetpack-module-search` are ok.

Since the second one returns a list of sites, it could also be something like `jetpack-sites-with adwords on`.

In the end, users we'll get used to whatever name is chosen. We could also discuss creating a new issue to revisit the naming of all existing commands, if the team agrees (for instance, I really want to change `remove-user` to `offboard-collaborator`)

NickGreen commented:
Tested. LGTM. I have no feedback on this one.

nickpagz commented:
@NickGreen I've updated the code based on your comments. Instead of the quote I've made it more useful by indicating how long the process may take and how to abort.
I've also changed the command to `jetpack-sites-with`, and added validation for both the module name and the status.

ecairol commented:
@nickpagz IMO you can merge this now. cc @NickGreen any thoughts?

NickGreen commented:
LGTM, go for it!
